[VEG-694] app creation and updates for products other than Support

token-cjg opened the pull request:
@zendesk/vegemite 
<!-- structure the Title above as the first line of a
     https://conventionalcommits.org/ message. example: "feat(buttons):
     add a muted button component". the title informs the semantic
     version bump if this PR is merged. -->

## Description

Adds support for app creation and app updates for products other than Support, i.e. for Chat and Sell.

Downgrades the `oclif/plugin-help` package from an earlier Snyk change as it was causing zcli to break.

## Checklist

- [ ] :guardsman: includes new unit and functional tests

anushkafka reviewed (approved):
This looks great, hope its been tested with chat only account as well 👍

sarmadsangi commented on packages/zcli-apps/src/commands/apps/create.ts:
```diff
@@ -53,7 +53,8 @@ export default class Create extends Command {
 
         const settings = manifest.parameters ? await getAppSettings(manifest, configParams) : {}
-        const installed = await request.requestAPI('api/v2/apps/installations.json', {
+        const product = Object.keys(manifest.location)[0]
```
Wondering if we need to consider multiple locations use-case here? ie 
```
  "location": {
    "support": {
      "ticket_editor": "assets/iframe.html",
      "nav_bar": "assets/iframe.html"
    },
    "sell": {
      "top_bar": "assets/iframe.html"
    }
  },
```

token-cjg replied:
A good point - although I'm not sure that zcli supported multiple locations prior to this change.  I can certainly look into this though, as this is the sort of feature that we probably should have.

sarmadsangi commented:
>Downgrades the oclif/plugin-help package from an earlier Snyk change as it was causing zcli to break.

My bad, i assumed our tests would catch a package breaking issue, if it did not then perhaps we need to explore more on why not.

sarmadsangi commented on packages/zcli-apps/src/commands/apps/update.ts:
```diff
@@ -35,21 +35,30 @@ export default class Update extends Command {
+      const updatedSummary: string[] = []
 
-      const installations: Installations = await request.requestAPI('/api/v2/apps/installations.json', {}, true)
+      Object.keys(manifest.location).forEach(async product => {
```
So we are looping through locations, is this how ZAT does it? My suggestion would be to see if we can do it via single endpoint instead of looping through per product and making multiple calls. But if this is the only way to do it then i guess no choice. 

cc @fbvilela (for ZAM review since they own these endpoints and might have some recommendations)

token-cjg replied:
Yes, ZAT loops through locations, see [here](https://github.com/zendesk/zendesk_apps_tools/blob/master/lib/zendesk_apps_tools/command.rb#L201-L204).  It appears that this is the recommended approach by the ZAM group, so closing this thread per said discussion.

cl8au commented on packages/zcli-apps/src/commands/apps/update.ts:
```diff
@@ -35,21 +35,30 @@ export default class Update extends Command {
     try {
       const { app_id }: any = await getUploadJobStatus(job_id, appPath)
       cli.action.stop('Deployed')
+      const updatedSummary: string[] = []
```
nit: I guess we can extract the following code out to be a function and only return a boolean if you need. Then you should be early return in the Object.keys.forEach easily. And you don't need to concat the string and filter the string at the end.

token-cjg replied:
Addressed by https://github.com/zendesk/zcli/pull/75/commits/1fc8dc0ffde660ced67bbf6af8a20e7eb0a1d99e

cl8au commented on packages/zcli-apps/src/commands/apps/create.ts:
```diff
@@ -53,15 +53,22 @@ export default class Create extends Command {
-          body: JSON.stringify({ app_id: `${app_id}`, settings: { name: manifest.name, ...settings } }),
-          headers: {
-            'Content-Type': 'application/json'
+        const installedSummary: string[] = []
```
nit: same as below :)

token-cjg replied:
Addressed by https://github.com/zendesk/zcli/pull/75/commits/8802f685823ec2d0add411be529b57c7bde3725f

token-cjg commented:
@cl8au, thanks for your review.  I've had a go at addressing your comments, let me know if all looks good, and, if so, I will move to ship this change.
